app/features/worksheet/tools.py

```diff
@@ -0,0 +1,278 @@
+            return {"message": "error", "data": "Number of Worksheets cannot exceed 10"}
+        
+        summary_obj = Summary(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_summary, self.parser_summary, self.verbose)
+        fill_in_the_blank_obj = FillInTheBlankQuestion(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_fill_in_blank, self.parser_fill_in_blank, self.verbose)
+        multiple_choice_obj = MultipleChoiceQuestion(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_multiple_choice, self.parser_multiple_choice, self.verbose)
+        open_ended_obj = OpenEndedQuestion(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_open_ended, self.parser_open_ended, self.verbose)
+        true_false_obj = TrueFalseQuestion(self.model, self.embedding_model, self.topic, self.grade_level, self.prompt_true_false, self.parser_true_false, self.verbose)
+        generated_worksheets = []
+        if self.verbose: logger.info(f"Creating Summary")
+        generated_summary = summary_obj.create_questions(num_questions = 1)
+        if generated_summary:
+            generated_summary = generated_summary[0]
+        else:
+            generated_summary = {'description': ''}
+        for i in range(num_worksheets):
+            worksheet = {}
+            worksheet.update(generated_summary)
+            if self.verbose: logger.info(f"Creating {num_fill_in_blank} Fill-in-the-blank questions")
+            generated_fill_in_blank = fill_in_the_blank_obj.create_questions(num_questions = num_fill_in_blank)
+            worksheet["fill_in_blank"] = generated_fill_in_blank
+            if self.verbose: logger.info(f"Creating {num_multiple_choice} Multiple-choice questions")
+            generated_multiple_choice = multiple_choice_obj.create_questions(num_questions = num_multiple_choice)
+            worksheet["multiple_choice"] = generated_multiple_choice
+            if self.verbose: logger.info(f"Creating {num_open_ended} Open-ended questions")
+            generated_open_ended = open_ended_obj.create_questions(num_questions = num_open_ended)
+            worksheet["open_ended"] = generated_open_ended
+            if self.verbose: logger.info(f"Creating {num_true_false} True or False questions")
+            generated_true_false = true_false_obj.create_questions(num_questions = num_true_false)
+            worksheet["true_false"] = generated_true_false
+
+            generated_worksheets.append(worksheet)
+        # Return the list of worksheets
+        if self.verbose: logger.info(f"Created {len(generated_worksheets)} Worksheets")
+        return generated_worksheets
+
+
+class QuestionChoiceFormat(BaseModel):
+    key: str = Field(description="A unique identifier for the choice using letters A, B, C, D, etc.")
+    value: str = Field(description="The text content of the choice")
+class MultipleChoiceQuestionFormat(BaseModel):
```

Comment:
For more info. about input validation, check this doc: https://docs.pydantic.dev/latest/concepts/validators/#ordering-of-validators-within-annotated